### ISSUE TITLE
Forward all system environment variables automatically?

### ISSUE THREAD

**Issue by klinquist** [Unitech/pm2#867]
Is there any way to forward everything in /etc/environment to PM2 processes?

Thanks!

**Comment by RichardJECooke** [comment]
I think I'm getting this same issue. When I run my app through pm2, **sudo pm2 start app.js** it can't see my environment variable, **$db**.  But if I run my app using **node app.js** it sees the environment variable fine, or if I **echo $db**.

This thread is very long. Can someone tell me if the bug is fixed yet? I'm not sure how to download the patch.

**Comment by bennettatoms** [comment]
I just ran into this issue on ubuntu 14.04 , node 6.9.1. My issue arose when a running server went down and a new one was brought back up without my application code deployed on it (but with a couple environment variables exported into `/etc/environment`). So I deployed my code to the new server (at the end of which a script calls `pm2 start`) but with only the aforementioned environment variables in place. So I manually entered the necessary additional environment variables in `/etc/environment`, logged out of the server and back in, then ran `pm2 restart all`, and all the newly added environment variables were showing up `undefined` when I logged them to the console. 

1. I tried adding them to `~/.bash_profile` and running `source ~/.bash_profile`, but that didn't work.

2. I tried adding the variables to `DEFAULT env_keep="..."` in `/etc/sudoers`, logged out and back in, but that didn't work. 

3. Finally, I ran `pm2 stop all`, noted the id of the process for the application (0), deleted the application process with `pm2 delete 0`, then ran `pm2 start server.js...` from scratch and the variables were logged correctly.

Anyway, I hope this helps someone else in the future.
